compass_reader/compass.py

```diff
@@ -0,0 +1,88 @@
+import threading
+import cv2
+from cv2 import aruco
+import numpy as np
+import time
+import math
+
+COMPASS_ID = 17
+BUFFER_SIZE = 5  # Number of values before giving a result
+WEBCAM_ID = 0
+DEBUG = True
+
+class CompassWatcher:
+    def __init__(self):
+        # Error
+        self.error = None
+
+        # Compass values
+        self.values = ""
+
+        # Aruco config
+        self._aruco_dict = aruco.Dictionary_get(aruco.DICT_4X4_250)
+        self._aruco_params = aruco.DetectorParameters_create()
+
+        # Start `run` in a separate thread
+        self.should_stop = False
+        thread = threading.Thread(target=self.run)
+        thread.daemon = True
+        thread.start()
+
+    def run(self):
+        cap = cv2.VideoCapture(WEBCAM_ID)
+
+        while(self.should_stop == False):
```

Comment:
`while(not self.should_stop)` is prettier here.

Reply:
I've tried with `while(!self.should_stop)` 😅 thanks!